### ISSUE TITLE
[Key Vault] "No content provided" error when calling `KeyVaultSettingsClient.update_setting`

### ISSUE THREAD

**Issue by mccoyp** [Azure/azure-sdk-for-python#30382]
**Behavior:** If a new KeyVaultSettingsClient is created and tries to call the `update_setting` method, the service responds with a "No content provided" error. If a different client request is successfully made first -- for example, calling `get_setting` -- `update_setting` will then work as expected.

**Affected package:** `azure-keyvault-administration` version 4.3.0.

**Cause:** The requests that return this error are missing an `Authorization` header and any body content (the setting's new value should be specified in the body). Requests succeed after the client successfully calls another method because a valid access token is obtained for the latter service call; this token is then used in subsequent `update_setting` calls successfully.

This suggests that the authentication policy fails to correctly obtain an access token if `update_setting` is invoked. `update_setting` is the only client method that uses PATCH requests in `azure-keyvault-administration`, so it's possible that this bug is caused by the SDK improperly handling PATCH requests. However, PATCH requests are used by client methods in other Key Vault libraries, so it's possible that a service bug is involved.

**Comment by mccoyp** [comment]
This seems to actually be a service issue. Managed HSM isn't returning a 401 response and authentication challenge when an unauthenticated PATCH request is sent to the "{vault URI}/settings/" endpoint.

For example, this afternoon I tried sending a setting update request with a fresh client. Our authentication policy identified that we didn't have a cached challenge for the Managed HSM endpoint, so we followed the usual protocol: sending an unauthenticated, empty request to the request endpoint.

Instead of returning a 401 response including an authentication challenge, the service returned the 400 "No content provided" response described in this issue.

In comparison, I tried doing a similar test with a different KV client method that uses PATCH requests (`KeyClient.update_key_properties`, from `azure-keyvault-keys`). The authentication policy triggered an unauthenticated, empty request, but this time we got the expected 401 challenge.

If I had to guess, it seems like the "{vault URI}/settings/" endpoint's error handling is responding to the empty request content before checking for request authentication. This seems to be unique; I haven't seen this type of error with any other requests.
